[ios] [expo-ui] Allow lower case section titles

Pflaumenbaum opened the pull request:
# Why

Because section titles are automatically capitalised and there was no option to make them lowercase

# How

By setting textCase to nil when the displayTitleUppercase property is set to false.

# Test Plan


<img width="435" alt="lowercase-title" src="https://github.com/user-attachments/assets/67d62420-ed95-4a22-a30c-4d4f12445b4f" />



# Checklist


- [x] I added a `changelog.md` entry and rebuilt the package sources according to [this short guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

ALBUS12345 committed:
[expo-ui] allow lowercase sectiontitles

Pflaumenbaum committed:
Update CHANGELOG.md

Pflaumenbaum commented:
@behenate  I have created a new pull request because the previous one had too many commits and became cluttered. This new PR includes only the relevant changes, and removes unnecessary commits for clarity.

aleqsio commented:
Let's not do this as a separate prop - i think we can assume that users will capitalize the strings manually instead and have preserve case as default.

Pflaumenbaum committed:
Always use lower case for section titles.

Pflaumenbaum commented:
@aleqsio I’ve modified it so that the title is automatically displayed in lowercase.

aleqsio commented:
Looks good! 
Maybe we can also add a comment to the prop that suggests expected capitalisation by the OS?

ALBUS12345 committed:
fix SwiftLint + add comment about platform conventions and use SectionProps in .tsx and .ios.tsx

Pflaumenbaum commented:
@aleqsio I've added the comment to the title prop in JSDoc. I also moved the SectionProps type to the index.tsx file and imported it to the index.ios.tsx file. Also, the web version now uses the SectionProps type instead of a custom one. 

https://github.com/expo/expo/pull/35113/commits/5ad291124aebe7491e93b5ea1109b9150cd070f4

Pflaumenbaum commented:
@behenate @aleqsio Are there any other changes I should make to take this further? I think I have made all the changes you suggested.

aleqsio commented:
Hi @Pflaumenbaum – sorry for not landing this earlier – this is mostly ready to go – if you can rebase it (we removed the polyfill on Android) happy to land this ASAP.

Pflaumenbaum committed:
rebase index.tsx

Pflaumenbaum committed:
tsbuilt

Pflaumenbaum commented:
@aleqsio I have rebased it and I think it's ready.
